### PR TITLE
sync with release 81 for HipPy code

### PR DESCRIPTION
**Pull request by meng-xiao**
enable HipPy for multiple IOV and reweighting datasets

### CODE REVIEW

**Comment by mmusich** [comment]
please test

File: Alignment/HIPAlignmentAlgorithm/src/HIPAlignmentAlgorithm.cc [hunk 1]
```diff
@@ -107,16 +123,55 @@ HIPAlignmentAlgorithm::initialize( const edm::EventSetup& setup,
...
+//      << "\nTrying to apply "<< setup.get<GlobalPositionRcd>().key().name()
+//      << " with multiple IOVs in tag.\n"
+//      << "Validity range is "
+//      << iov.first().eventID().run() << " - " << iov.last().eventID().run();
+//  }
+
+//  const Rcd & record = setup.get<TrackerAlignmentRcd>();
+  const edm::ValidityInterval & validity = setup.get<TrackerAlignmentRcd>().validityInterval();
+  const edm::IOVSyncValue first1 = validity.first();
+	std::cout << "xiaomeng "<<first1.eventID().run()<<std::endl;
```
Comment (mmusich) [inline Alignment/HIPAlignmentAlgorithm/src/HIPAlignmentAlgorithm.cc]: @meng-xiao would you please replace `cout` with usage of the messageLogger service?

File: Alignment/HIPAlignmentAlgorithm/python/common_cff_py.txt [hunk 2]
```diff
@@ -1,8 +1,25 @@
 from CondCore.DBCommon.CondDBSetup_cfi import *
 
 # loading magnetic field and geometry
+
+#process.load('Configuration.Geometry.GeometryExtended_cff')
+#process.TrackerTopologyEP = cms.ESProducer("TrackerTopologyEP")
```
Comment (mmusich) [inline Alignment/HIPAlignmentAlgorithm/python/common_cff_py.txt]: @meng-xiao there are many commented lines with obsolete configuration. Is it possible to clean them up?

**Comment by mmusich** [comment]
@meng-xiao do we expect further work on this PR? there are a few unanswered comments

**Comment by mmusich** [comment]
@meng-xiao ping

**Comment by ghellwig** [comment]
@meng-xiao what's the status of this PR? Please address the above comments.

**Comment by mmusich** [comment]
-1
superseded by: https://github.com/cms-sw/cmssw/pull/14674

**Pull request closed without merging**